plugins/outputs/groundwork/groundwork.go

```diff
@@ -0,0 +1,349 @@
+			return fmt.Errorf("re-login failed: %v", err)
+		}
+		headers["GWOS-API-TOKEN"] = g.authToken
+		statusCode, body, err = clients.SendRequest(http.MethodPost, g.Server+defaultMonitoringRoute, headers, nil, requestJSON)
+		if err != nil {
+			return fmt.Errorf("error while sending: %v", err)
+		}
+	}
+
+	if statusCode != 200 {
+		return fmt.Errorf("HTTP request failed. [Status code]: %d, [Response]: %s", statusCode, string(body))
+	}
+
+	return nil
+}
+
+func (g *Groundwork) Description() string {
+	return "Send telegraf metrics to GroundWork Monitor"
+}
+
+func init() {
+	outputs.Add("groundwork", func() telegraf.Output {
+		return &Groundwork{
+			ResourceTag:         "host",
+			DefaultHost:         "telegraf",
+			DefaultServiceState: string(transit.ServiceOk),
+		}
+	})
+}
+
+func (g *Groundwork) parseMetric(metric telegraf.Metric) (string, transit.DynamicMonitoredService) {
+	resource := g.DefaultHost
+	if value, present := metric.GetTag(g.ResourceTag); present {
+		resource = value
+	}
+
+	service := metric.Name()
+	if value, present := metric.GetTag("service"); present {
+		service = value
+	}
+
+	status := g.DefaultServiceState
+	value, statusPresent := metric.GetTag("status")
+	if validStatus(value) {
+		status = value
+	}
+
+	message, _ := metric.GetTag("message")
+
+	unitType := string(transit.UnitCounter)
+	if value, present := metric.GetTag("unitType"); present {
+		unitType = value
+	}
+
+	critical := -1.0
+	value, criticalPresent := metric.GetTag("critical")
+	if criticalPresent {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			critical = s
+		}
+	}
+
+	warning := -1.0
+	value, warningPresent := metric.GetTag("warning")
+	if warningPresent {
+		if s, err := strconv.ParseFloat(value, 64); err == nil {
+			warning = s
+		}
+	}
+
+	lastCheckTime := transit.NewTimestamp()
+	lastCheckTime.Time = metric.Time()
+	serviceObject := transit.DynamicMonitoredService{
+		BaseTransitData: transit.BaseTransitData{
+			Name:  service,
+			Type:  transit.Service,
+			Owner: resource,
+		},
+		Status:           transit.MonitorStatus(status),
+		LastCheckTime:    lastCheckTime,
+		LastPlugInOutput: message,
+		Metrics:          nil,
+	}
+
+	for _, value := range metric.FieldList() {
+		var thresholds []transit.ThresholdValue
+		if warningPresent {
+			thresholds = append(thresholds, transit.ThresholdValue{
+				SampleType: transit.Warning,
+				Label:      value.Key + "_wn",
+				Value: &transit.TypedValue{
+					ValueType:   transit.DoubleType,
+					DoubleValue: warning,
+				},
+			})
+		}
+		if criticalPresent {
+			thresholds = append(thresholds, transit.ThresholdValue{
+				SampleType: transit.Critical,
+				Label:      value.Key + "_cr",
+				Value: &transit.TypedValue{
+					ValueType:   transit.DoubleType,
+					DoubleValue: critical,
+				},
+			})
+		}
+
+		typedValue := new(transit.TypedValue)
+		err := typedValue.FromInterface(value.Value)
+		if err != nil {
+			typedValue = nil
+			g.Log.Errorf("%v", err)
```

Comment:
I would error out in this case. The metric could not be understood, so I'm assuming that's a hard break, no?

Reply:
Done